File: 2022_c#_basics/12-while-loop-exercise/02-exam-preparation.cs
using System;

namespace _02_Exam_Preparation
{
    internal class Program
    {
        static void Main(string[] args)
        {
            double unsatisfyingProblemsNumber = double.Parse(Console.ReadLine());
            double unsatisfyingProblemsNumberCount = 0;
            double allProblems = 0;
            double avg = 0;
            string nameProblem = Console.ReadLine();

            string lastProblem = string.Empty;

            while (nameProblem != "Enough")
            {
                lastProblem = nameProblem;
                double ratingProblem = double.Parse(Console.ReadLine());
                avg += ratingProblem;
                allProblems += 1;

                if (ratingProblem <= 4)
                {
                    unsatisfyingProblemsNumberCount += 1;
                }

                if (unsatisfyingProblemsNumber == unsatisfyingProblemsNumberCount)
                {
                    Console.WriteLine($"You need a break, {unsatisfyingProblemsNumberCount} poor grades.");
                    break;
                }

                nameProblem = Console.ReadLine();
            }

            if (nameProblem == "Enough")
            {
                double averageScore = avg / allProblems;

                Console.WriteLine($"Average score: {averageScore:f2}");
                Console.WriteLine($"Number of problems: {allProblems}");
                Console.WriteLine($"Last problem: {lastProblem}");

            }
        }
    }
}

File: 2024_java_basics/12-while-loops-lab/tasks/02-exam-preparation.java
import java.util.Scanner;

public class ExamPreparation {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        // Read the maximum number of unsatisfactory problems allowed
        double unsatisfyingProblemsNumber = Double.parseDouble(scanner.nextLine());
        double unsatisfyingProblemsNumberCount = 0;
        double allProblems = 0;
        double totalRating = 0;
        String nameProblem = scanner.nextLine();

        String lastProblem = "";

        while (!nameProblem.equals("Enough")) {
            lastProblem = nameProblem;
            double ratingProblem = Double.parseDouble(scanner.nextLine());
            totalRating += ratingProblem;
            allProblems++;

            if (ratingProblem <= 4) {
                unsatisfyingProblemsNumberCount++;
            }

            if (unsatisfyingProblemsNumber == unsatisfyingProblemsNumberCount) {
                System.out.println("You need a break, " + (int) unsatisfyingProblemsNumberCount + " poor grades.");
                break;
            }

            nameProblem = scanner.nextLine();
        }

        if (nameProblem.equals("Enough")) {
            double averageScore = totalRating / allProblems;

            System.out.printf("Average score: %.2f%n", averageScore);
            System.out.println("Number of problems: " + (int) allProblems);
            System.out.println("Last problem: " + lastProblem);
        }

        scanner.close();
    }
}
